packages/extension-core/src/domains/analytics/store.ts

```diff
@@ -0,0 +1,142 @@
+import { settingsStore } from "../app/store.settings"
+import { PostHogCaptureProperties } from "./types"
+
+export type AnalyticsEvent = {
+  eventName: string
+  properties?: PostHogCaptureProperties
+  timestamp: number
+}
+
+type AnalyticsData = { data: AnalyticsEvent[]; distinctId: string }
+
+const TALISMAN_PROPERTIES = {
+  appVersion: process.env.VERSION,
+  appBuild: process.env.BUILD,
+  testBuild: DEBUG || ["dev", "qa", "ci"].includes(process.env.BUILD as string),
+}
+
+class AnalyticsStore extends StorageProvider<AnalyticsData> {
+  // the isReady promise prevents events being added to the queue while the analytics are being sent
+  isReady: Promise<boolean> = new Promise(() => true)
+
+  constructor(name: string) {
+    super(name)
+    setInterval(async () => await this.send(), 60_000)
+  }
+
+  async getDistinctId() {
+    try {
+      // Identify the posthog client
+      //
+      // We have to do this manually, because we don't have access to `localStorage` in the background script,
+      // and `localStorage` is the only built-in way for posthog to persist the distinct_id.
+      let posthogDistinctId = await this.get("distinctId")
+
+      // if this Talisman instance doesn't already have a persisted distinct_id, randomly generate one
+      if (posthogDistinctId === undefined) {
+        posthogDistinctId = v4()
+        await this.mutate((data) => ({ ...data, distinctId: posthogDistinctId }))
+      }
+    } catch (cause) {
+      const error = new Error("Failed to identify posthog client", { cause })
+      console.error(error) // eslint-disable-line no-console
+      Sentry.captureException(error)
+    }
+  }
+
+  async capture(
+    eventName: string,
+    rawProperties?: PostHogCaptureProperties,
+    eventTimestamp?: number
+  ) {
+    log.debug("AnalyticsStore.capture", { eventName, rawProperties, eventTimestamp })
+    const timestamp = eventTimestamp ?? Date.now()
+    const properties = {
+      ...rawProperties,
+      ...TALISMAN_PROPERTIES,
+      distinct_id: await this.getDistinctId(),
+    }
+
+    await this.isReady
+    await this.mutate(({ data, distinctId }) => ({
+      distinctId,
+      data: [...data, { eventName, properties, timestamp }],
+    }))
+  }
+
+  captureDelayed(eventName: string, properties?: PostHogCaptureProperties, delaySeconds = 1800) {
+    // add a random delay of up to 30 min to avoid deanonymisation of user events
+    const timestamp = Date.now() + delaySeconds * 1000 * Math.random()
+    this.capture(eventName, properties, timestamp)
+  }
+
+  async send() {
+    const enabled = await settingsStore.get("useAnalyticsTracking")
+    if (!enabled) return
+    log.debug("AnalyticsStore.send")
+
+    await this.isReady
+
+    const sendInner = async () => {
+      if (!process.env.POSTHOG_AUTH_TOKEN) {
+        log.warn("POSTHOG_AUTH_TOKEN is not set, not sending analytics")
+        return true
+      }
+
+      const { data } = await this.get()
```

Comment:
When testing, data here is undefined eventhough i captured a lot of events up front, so it stops there and never sends anything to posthog